hazelcast/test/src/HazelcastTests3.cpp

```diff
@@ -999,6 +1020,9 @@ class BasicClientReplicatedMapNearCacheTest
     {
         client_config nearCachedClientConfig = get_config();
         nearCachedClientConfig.add_near_cache_config(near_cache_config_);
+        if (near_cached_client_ != nullptr) {
+            near_cached_client_->shutdown().get();
+        }
```

Comment:
These lines are OK but latest `near_cached_client_` instance will be left as is. So destructor is needed.